src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -103,14 +104,32 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
 
     case n@PForStmt(_, cond, _, _, _) => isExpr(cond).out ++ comparableTypes.errors(exprType(cond), BooleanT)(n)
 
-    case n@PShortForRange(exp, lefts, _) =>
-      if (lefts.forall(pointsToData)) multiAssignableTo.errors(Vector(miscType(exp)), lefts map idType)(n)
-      else error(n, s"at least one assignee in $lefts points to a type")
-
-
-    case n@PAssForRange(exp, lefts, _) =>
-      multiAssignableTo.errors(Vector(miscType(exp)), lefts map exprType)(n) ++
-        lefts.flatMap(t => addressable.errors(t)(t))
+    case _@PShortForRange(range, shorts, _, _) =>
+      if (shorts.forall(pointsToData)) {
+        val typ = miscType(range).asInstanceOf[Type.InternalTupleT]
+        if (shorts.length > 0 && shorts.length <= typ.ts.length) {
+          if (typ.ts.take(shorts.length).zip(shorts map idType).forall(assignableTo))
+            noMessages
+          else error(range, "Can not assign range types to identifiers")
+        }
+        else error(range.exp, "Wrong number of identifiers for range")
+      }
+      else error(range.exp, s"at least one assignee in $shorts points to a type")
```

Comment:
what is this supposed to mean?

Reply:
It was there before I started altering the type checking and I thought it was needed, after further inspection it is not, removed.